packages/core-data/src/private-actions.js

```diff
@@ -14,3 +24,114 @@ export function receiveRegisteredPostMeta( postType, registeredPostMeta ) {
 		registeredPostMeta,
 	};
 }
+
+/**
+ * @typedef {Object} Modifier
+ * @property {string} [type] - The type of modifier.
+ * @property {Object} [args] - The arguments of the modifier.
+ */
+
+/**
+ * @typedef {Object} Edits
+ * @property {string}     [src]       - The URL of the media item.
+ * @property {Modifier[]} [modifiers] - The modifiers to apply to the media item.
+ */
+
+/**
+ * Duplicates a attachment / media entity record and, optionally, modifies it.
+ *
+ * @param {string}   kind                    Entity kind.
+ * @param {string}   name                    Entity name.
+ * @param {string}   recordId                Entity record ID.
+ * @param {Edits}    edits                   Edits to apply to the record.
+ * @param {Object}   options                 Options object.
+ * @param {Function} options.__unstableFetch Custom fetch function.
+ * @param {boolean}  options.throwOnError    Whether to throw an error if the request fails.
+ *
+ * @return {Promise} Promise resolving to the updated record.
+ */
+export const editMediaEntity =
+	(
+		kind,
+		name,
```

Comment:
Because the action is specifically for media, the kind and name perhaps don't need to be specifiable. We don't want folks to call the action for other non-media entities.

Reply:
I'm working on deprecating one of attachment or media (probably media by the looks of it) - https://github.com/WordPress/gutenberg/pull/70833.

But having said that, I think it might be best to hardcode it to use root/media (it's what's used elsewhere in gutenberg), and then when I deprecate I can switch this function over to use whatever the stable entity is.

Reply:
Done in [3149e64](https://github.com/WordPress/gutenberg/pull/70814/commits/3149e64086e8763a24b0a90ab3379506fbc6fb25)